### ISSUE TITLE
Using undeclare() on a numeric type with char missing reasons results in a confusing type

### ISSUE THREAD

**Issue by khusmann** [dusadrian/declared#9]
Not sure if this is expected behavior or not, but this behavior was confusing / surprising to me. Given a numeric type with char missing reasons:

```r
(x <- declared(c(1,2,"a"), na_values="a"))
#> <declared<numeric>[3]>
#> [1]     1     2 NA(a)
#> Missing values: a
```

When I `undeclare(x)`, it still reports it's a numeric type:

```r
undeclare(x)
#> <declared<numeric>[3]>
#> [1] 1 2 a
```

Even though `is.numeric()` and `is.character` report otherwise:

```r
is.numeric(undeclare(x))
#> [1] FALSE
is.character(undeclare(x))
#> [1] TRUE
```

Should `undeclare(x)` instead be printing that it's a `<declared<character>[3]>` instead?

**Comment by dusadrian** [comment]
This seems like a genuine bug, will start fixing it, thanks.

**Comment by dusadrian** [comment]
Fixed in 167d2b95585448c7c8646922249728be6a5a68d1
